### PR TITLE
Add checkstyle configuration

### LINKED ISSUE #1167: Review and standardize checkstyle rules
As a developer, I'd like to review and standardize checkstyle rules, so that I can apply the rules across all the SCDF ecosystem of projects for consistency.

**Acceptance:**
Categorize the current rules in the context of noise vs. usefulness. For example, any deprecated use of APIs should fail the build.

### LINKED ISSUE #1226: Review eclipse/intellij formatting files
As a developer, I'd like to review SCDF's Eclipse/IntelliJ formatting files, so I could adjust it to align with Spring's guideline.

**Acceptance:**
- The goal is to be consistent across all the projects in SCDF ecosystem.
- Add the formatting files to all projects in the ecosystem.

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Add support for running checkstyle plugin
  - Add basic rules at this phase

Resolves #1167
Resolves #1226

### CODE REVIEW

File: checkstyle.xml [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        <module name="ParameterAssignment"/>
+        -->
+        <!-- Imports -->
+        <!--<module name="AvoidStarImport"/>-->
+        <!--<module name="AvoidStaticImport">-->
+            <!--<property name="excludes"-->
+                      <!--value="org.junit.Assert.*,org.mockito.Mockito.*,org.mockito.Matchers.*,org.hamcrest.Matchers.*,org.assertj.core.api.Assertions.*"/>-->
+        <!--</module>-->
+        <!--<module name="FallThrough"/>-->
+        <!--<module name="ImportOrder">-->
```
Comment (mbogoevici) [inline checkstyle.xml]: We might want to enable this, too.

**Comment by ilayaperumalg** [comment]
Rebased and fixed the checkstyle config exceptions for the newly added files.

File: checkstyle.xml [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+            <!--<property name="max" value="1"/>-->
+            <!--<property name="tokens" value="LAMBDA"/>-->
+        <!--</module>-->
+        <!--
+        <module name="ReturnCount">
+            <property name="max" value="3"/>
+            <property name="tokens" value="METHOD_DEF"/>
+        </module>
+        -->
+        <!--<module name="UnusedImports"/>-->
```
Comment (markpollack) [inline checkstyle.xml]: this should be enabled...

File: README.adoc [hunk 3]
```diff
@@ -62,5 +62,22 @@ For more information on building, see this https://github.com/spring-cloud/sprin
 
 We welcome contributions! Follow this https://github.com/spring-cloud/spring-cloud-dataflow/blob/master/spring-cloud-dataflow-docs/src/main/asciidoc/appendix-contributing.adoc[link] for more information on how to contribute.
 
-
-
+=== Code formatting guidelines
+
+* Use the Spring Boot code formatter from https://github.com/spring-projects/spring-boot/blob/master/eclipse/eclipse-code-formatter.xml[here]
+* If you are using `IntelliJ` then Go to Preferences -> Editor -> Code Style -> Java -> Manage and import the formatter file
```
Comment (markpollack) [inline README.adoc]: Go to `File->Settings->Editor....`

File: README.adoc [hunk 4]
```diff
@@ -62,5 +62,22 @@ For more information on building, see this https://github.com/spring-cloud/sprin
...
+
+  ** Go to Import settings in Preferences -> Editor -> Code Style -> Java -> Import and set
```
Comment (markpollack) [inline README.adoc]: `Setting -> Editor -> Code Style -> Java` then select the `Imports` tab.

Comment (ilayaperumalg) [inline README.adoc]: I think these steps varies from IntelliJ version and OS.

File: README.adoc (hunk 3 above)
Comment (ilayaperumalg) [inline README.adoc]: Same here, these steps varies for different versions and OS.

File: README.adoc [hunk 5]
```diff
@@ -62,5 +62,22 @@ For more information on building, see this https://github.com/spring-cloud/sprin
...
+  `Class count to use import with '\*'` and `Names count to use static import with '*'` to high values (say 999) to avoid using this property
+  ** In the same page, set the Import order with the following settings
```
Comment (markpollack) [inline README.adoc]: set the `Import Layout`

**Comment by markpollack** [comment]
Additional commits on top of this one as part of da2a1bccac25a13aaa090ccf37056af740e96654

* Moved checkstyle files into a subdirectory
* Various tweaks to checkstyle.xml
* Reformatted code base in IntelliJ

**Pull request closed without merging**